Makefile

```diff
@@ -88,93 +88,97 @@ OUT_TESTS      = $(BINDIR)/sirtests$(PLATFORM_EXE_EXT)
 # targets
 # ##########
 
+.DEFAULT_GOAL := all
 .PHONY: all
 all: $(OUT_SHARED) $(OUT_STATIC) $(OUT_EXAMPLE) $(OUT_TESTS)
 
 -include $(INTDIR)/*.d
 
+.RECIPEPREFIX = ~
+
 $(OBJ_EXAMPLE): $(EXAMPLE)/$(EXAMPLE).c $(DEPS)
-	@mkdir -p $(@D)
-	$(CC) $(MMDOPT) -c -o $@ $< $(CFLAGS) -I..
+~  @mkdir -p $(@D)
+~  $(CC) $(MMDOPT) -c -o $@ $< $(CFLAGS) -I..
 
 $(OBJ_TESTS): $(TESTS)/$(TESTS).c $(DEPS)
-	@mkdir -p $(@D)
-	$(CC) $(MMDOPT) -c -o $@ $< $(CFLAGS) -I..
+~  @mkdir -p $(@D)
+~  $(CC) $(MMDOPT) -c -o $@ $< $(CFLAGS) -I..
 
 $(INTDIR)/%.o: %.c $(DEPS)
-	@mkdir -p $(@D)
-	$(CC) $(MMDOPT) -c -o $@ $< $(CFLAGS)
+~  @mkdir -p $(@D)
+~  $(CC) $(MMDOPT) -c -o $@ $< $(CFLAGS)
 
 .PHONY: shared
 shared: $(OUT_SHARED)
 $(OUT_SHARED): $(OBJ_SHARED)
-	@mkdir -p $(@D)
-	$(CC) -shared -o $(OUT_SHARED) $^ $(CFLAGS) $(LDFLAGS_SHARED)
-	-@echo built $(OUT_SHARED) successfully.
+~  @mkdir -p $(@D)
+~  $(CC) -shared -o $(OUT_SHARED) $^ $(CFLAGS) $(LDFLAGS_SHARED)
+~  -@echo built $(OUT_SHARED) successfully.
 
 .PHONY: static
 static: $(OUT_STATIC)
 $(OUT_STATIC): $(OUT_SHARED)
-	@mkdir -p $(@D)
-	ar -cr $(OUT_STATIC) $(OBJ_SHARED)
-	-@($(RANLIB) "$(OUT_STATIC)" || true) > /dev/null 2>&1
-	-@echo built $(OUT_STATIC) successfully.
+~  @mkdir -p $(@D)
```

Comment:
Wait, what does the tilde do?

Reply:
Instead of tab for receipe prefix, which displays as either 2, 5, or 8 spaces depending on the terminal or editor, it sets `.RECIPEPREFIX = ~` which will always be a fixed width, and now everything can be aligned/indented to a predictable width, and without spaces vs. tabs being significant to syntax:

![image](https://github.com/aremmell/libsir/assets/61629094/3bf29321-0ca8-4307-b636-621c60a3d50c)

Reply:
Using `:` works too and looks good also:

![image](https://github.com/aremmell/libsir/assets/61629094/865a20b3-6bcc-4447-9dbe-ba273d9c275d)

Reply:
I like that better actually.

Reply:
Yeah, better.